### PR TITLE
fix: skip backend sync on force push

### PR DESCRIPTION
**Pull request by lazpavel**
#### Description of changes
— skips backend sync on force push, as force push is ignoring resource status and always executes the push
— this also helps to override deployed backed in case if the deployed backed zip is corrupted

#### Description of how you validated changes
— manual changes
— unit tests added
— `yarn test` and `yarn e2e` pass

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/commands/push.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { updateCognitoTrackedFiles } from '../extensions/amplify-helpers/update-
 /**
  * Download and unzip deployment bucket contents to #current-cloud-backend so amplify status shows correct state
  */
-const syncCurrentCloudBackend = async (context: $TSContext): Promise<void> => {
+export const syncCurrentCloudBackend = async (context: $TSContext): Promise<void> => {
```
Comment (Amplifiyer) [inline packages/amplify-cli/src/commands/push.ts:14]: I know this is being exported for assertions in the unit tests, but exporting functions that are expected to be 'internal' can cause problems in the future. Is there any way we can skip exporting this and updateTrackedFiles (e.g. you can mock `updateCognitoTrackedFiles` instead)

Comment (lazpavel) [reply packages/amplify-cli/src/commands/push.ts:14]: I did update the updateTrackedFiles, for syncCurrentCloudBackend I prefer to not test on inner methods calls as that might change and miss the point of the test, I did however moved in a separate module

File: packages/amplify-cli/src/commands/push.ts [hunk 2]
```diff
@@ -61,7 +20,12 @@ export const run = async (context: $TSContext): Promise<$TSAny> => {
   if (context.parameters.options?.force) {
     context.exeInfo.forcePush = true;
   }
-  await syncCurrentCloudBackend(context);
+
+  // force flag ignores project status check
+  if (!context.exeInfo.forcePush) {
+    await syncCurrentCloudBackend(context);
+  }
```
Comment (sobolk) [inline packages/amplify-cli/src/commands/push.ts:27]: Should we add an e2e test where we nuke deployment bucket and then attempt to force push ?

Comment (lazpavel) [reply packages/amplify-cli/src/commands/push.ts:27]: will do

**Review by sobolk — APPROVED** [review]

**Review by Amplifiyer — APPROVED** [review]

**Pull request merged**